### DISCUSSION TITLE
NonConstantInjection

### DISCUSSION THREAD

**Discussion by Josselin-76** [idaholab/moose#19276] [Q&A General] [answered]
Dear community,

Thank you for your wonderful work and your constant new features.

I am modeling a packer system with a non constant pressure injection rate into a fracture Network. To do so I have used different tools that allowed me a variation of the injection rate over times ( `FunctionDirichletBC` and  `PorousFlowSquarePulsePointSource` ), nevertheless none of them give me enough accuracy this is why I am using `PorousFlowPeacemanBorehole` and `PorousFlowPolyLineSink`. 

I am not able to find how to apply a non constant injection pressure using `PorousFlowPeacemanBorehole` and `PorousFlowPolyLineSink`. Can someone enlighten me ? 

Josselin Ouf
RWTH Aachen, Germany

**Comment by WilkAndy** [comment, marked as answer]
Both [PorousFlowPolyLineSink](https://mooseframework.inl.gov/source/dirackernels/PorousFlowPolyLineSink.html) and [PorousFlowPeacemanBorehole](https://mooseframework.inl.gov/source/dirackernels/PorousFlowPeacemanBorehole.html) have a `multiplying_var` input.  This multiplies the fluxes by your defined `AuxVariable`.  Hopefully this does what you want.  You could make a `FunctionAux` for the `multiplying_var`, for instance.

**Reply by Josselin-76** [reply]
Thanks for your answer, 

I have try to make `FunctionAux` for the `multiplying_var` in ` PorousFlowPeacemanBorehole`. However it does not offer a significant control on the injection rate even when I am using a really small value for the pipe length and radius. 

My goal is to change the injection rate over time, this change is in method 1 perfectly executed by `FunctionDirichletBC` but the same injection function is partly ignored by the second method ` PorousFlowPeacemanBorehole`. Indeed in the second method the curve is a Heaviside like function while one may expect variation through time. 

Thanks for your help.

### Method 1
[pre_inj_method1]
   type = FunctionDirichletBC
   boundary = 'INJ2'
   variable = pwater
  function = 'if(t<=100,0.25,if(t<=362.2,7.34-0.0054*(t-100),if(t<=671.8,6.3-0.00245*(t-362.2),if(t<=974.8,5.85-0.00102*(t-671.8),if(t<=1265.8,5.67-0.000481*(t-974.8),if(t<=2175.4,5.62+0.000396*(t-1265.8),if(t<=4000,5.98-5.35*e-4*(t-2175.4),5)))))))'
[]

Result 

![Method 1](https://user-images.githubusercontent.com/93592660/140780212-bba19cc6-ca53-435d-be49-eeee6f0220f0.jpg)
### Method 2
[injection_packer]
    type = FunctionAux
    variable = injection_packer
    function = packer
      [../]
 [Functions]
    [packer]
    type = ParsedFunction
    value = 'if(t<=100,0.25,if(t<=362.2,7.34-0.0054*(t-100),if(t<=671.8,6.3-0.00245*(t-362.2),if(t<=974.8,5.85-0.00102*(t-671.8),if(t<=1265.8,5.67-0.000481*(t-974.8),if(t<=2175.4,5.62+0.000396*(t-1265.8),if(t<=4000,5.98-0.000535*(t-2175.4),5)))))))'
  []
[]
  [DiracKernels]
 
    [bh]
      type = PorousFlowPeacemanBorehole
     variable = pwater
     SumQuantityUO = injected_mass
      point_file = INJ.bh
      fluid_phase = 0
      bottom_p_or_t = 7.3 #MPa
      unit_weight = '0 0 0' #fluid_density*gravitational_acceleration, Typical value for water
      function_of = pressure
      character = -1 # -1 source, 1 a sink
      use_mobility = true
      re_constant = 0.1594
    multiplying_var = injection_packer
[]

Result
![method2 (2)](https://user-images.githubusercontent.com/93592660/140780546-553ca49f-3b69-431c-b19f-e213bbbae254.jpg)
